vxquery-core/src/main/java/org/apache/vxquery/metadata/VXQueryIndexingOperatorDescriptor.java

```diff
@@ -0,0 +1,233 @@
+        final String collectionName = collectionPartitions[partition % collectionPartitions.length];
+        String collectionModifiedName = collectionName.replace("${nodeId}", nodeId);
+
+        return new AbstractUnaryInputUnaryOutputOperatorNodePushable() {
+            @Override
+            public void open() throws HyracksDataException {
+                appender.reset(frame, true);
+                writer.open();
+            }
+
+            @Override
+            public void nextFrame(ByteBuffer buffer) throws HyracksDataException {
+                fta.reset(buffer);
+
+                IPointable result = new TaggedValuePointable();
+
+                final UTF8StringPointable stringp = (UTF8StringPointable) UTF8StringPointable.FACTORY.createPointable();
+                final TaggedValuePointable nodep = (TaggedValuePointable) TaggedValuePointable.FACTORY
+                        .createPointable();
+
+                final ByteBufferInputStream bbis = new ByteBufferInputStream();
+                final DataInputStream di = new DataInputStream(bbis);
+                final SequenceBuilder sb = new SequenceBuilder();
+                final ArrayBackedValueStorage abvs = new ArrayBackedValueStorage();
+                final ArrayBackedValueStorage abvsFileNode = new ArrayBackedValueStorage();
+
+                String indexModifiedName;
+
+                if (collectionModifiedName.contains("hdfs://")) {
+                    throw new HyracksDataException("Indexing support for HDFS not yet implemented.");
+                } else {
+
+                    if (AbstractCollectionRule.functionCall.getFunctionIdentifier().equals(BuiltinFunctions
+                            .FN_BUILD_INDEX_ON_COLLECTION_1.getFunctionIdentifier())) {
+                        indexModifiedName = VXQueryCommons.INDEX_CENTRALIZER_UTIL.putIndexForCollection
+                                (collectionModifiedName);
+                        File collectionDirectory = new File(collectionModifiedName);
+
+                        //check if directory is in the local file system
+                        if (collectionDirectory.exists() && collectionDirectory.isDirectory()) {
+                            IndexConstructorUtil indexConstructorUtil = new IndexConstructorUtil();
+                            try {
+                                indexConstructorUtil
+                                        .evaluate(collectionModifiedName, indexModifiedName, result, stringp, bbis, di,
+                                                sb, abvs, nodeIdProvider, abvsFileNode, nodep, false, nodeId);
+                                XDMConstants.setTrue(result);
+                                FrameUtils.appendFieldToWriter(writer, appender, result.getByteArray(), result.getStartOffset(),
+                                        result.getLength());
+                            } catch (SystemException | JAXBException e) {
```

Comment:
Please clean up exception handling. Lets try to throw fewer errors. same for all index methods.